persist/fs/seek.go

```diff
@@ -179,6 +179,37 @@ func (s *seeker) Open(namespace ts.ID, shard uint32, blockStart time.Time) error
 		return err
 	}
 
+	// Mmap necessary files
+	mmapOptions := mmapOptions{
+		read: true,
+		hugeTLB: mmapHugeTLBOptions{
+			enabled:   s.opts.MmapEnableHugeTLB(),
+			threshold: s.opts.MmapHugeTLBThreshold(),
+		},
+	}
+	mmapResult, err := mmapFiles(os.Open, map[string]mmapFileDesc{
+		filesetPathFromTime(shardDir, blockStart, indexFileSuffix): mmapFileDesc{
+			file:  &indexFd,
+			bytes: &s.indexMmap,
+			// Index files are small enough that they don't warrant hugeTLB
+			options: mmapOptions,
+		},
+		filesetPathFromTime(shardDir, blockStart, dataFileSuffix): mmapFileDesc{
+			file:    &dataFd,
+			bytes:   &s.dataMmap,
+			options: mmapOptions,
+		},
+	})
+	if err != nil {
+		s.Close()
+		return err
+	}
+	if warning := mmapResult.warning; warning != nil {
+		s.opts.InstrumentOptions().Logger().Warnf(
+			"warning while mmaping files in seeker: %s", warning.Error())
+	}
+
+	// Setup digest readers
```

Comment:
Perhaps move the `...Reset()` calls and the `defer func() { ...Close()  }` stanza's above the `mmapFiles` call because it could return `if err != nil { ... }` from the `mmapFiles` call?

Reply:
good call